tests/tests/test_decommission.py

```diff
@@ -0,0 +1,75 @@
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import pytest
+
+from ..common_setup import standard_setup_one_client
+from ..MenderAPI import authentication, devauth
+from .mendertesting import MenderTesting
+
+
+class TestDecommission(MenderTesting):
+    """Tests the decommission and re authentication functionality"""
+
+    def test_decommission(self, standard_setup_one_client):
+        mender_device = standard_setup_one_client.device
+
+        # accept the device
+        devauth.accept_devices(1)
+        # list of authorized devices
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        # check that there is a authorized device
+        assert 1 == len(devices)
+        # check from client side that the authorization was successful
+        journal_info = self.get_device_journal(mender_device)
+        assert "Server authorization successful" in journal_info
+
+        # decommission the device
+        devauth.decommission(devices[0])
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        # check that there is no authorized devices after decommission
+        assert 0 == len(devices)
+        old_journal_info = journal_info
+        journal_info = self.get_device_journal(mender_device)
+        # check on the client side that no new authorization has happened
+        assert "Server authorization successful" not in self.remove_old_journal_info(
+            old_journal_info, journal_info
+        )
+
+        # accept the device again
+        devauth.accept_devices(1)
+        # check again that the device is authorized
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert 1 == len(devices)
+        old_journal_info = journal_info
+        journal_info = self.get_device_journal(mender_device)
+        assert "Server authorization successful" in self.remove_old_journal_info(
+            old_journal_info, journal_info
+        )
+
+    def get_device_journal(self, device):
+        return device.run("journalctl -u %s" % device.get_client_service_name())
+
+    # returns a substring without the old journal info
+    def remove_old_journal_info(self, old, new):
+        # first line is different in every journal, therefore it gets deleted.
+        old = old[old.index("\n") + 1 :]
```

Comment:
I understand what you are doing, but is little intuitive, isn't it?

An alternative solution could be to convert to array with `old.split('\n')`, drop the first line, and (if necessary) convert again to string with `'\n'.join(old)`. You could even not convert back to string, and just manipulate the arrays.

Take this as a weak suggestion, experiment if you want but the current solution is acceptable.

Reply:
I agree, it should be more intuitive what the code does. 
I changed it to a solution that uses split.